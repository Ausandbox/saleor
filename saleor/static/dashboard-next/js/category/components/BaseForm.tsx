import Button from "material-ui/Button";
import Card, { CardContent, CardActions } from "material-ui/Card";
import Typography from "material-ui/Typography";
import { withStyles, WithStyles } from "material-ui/styles";
import * as React from "react";
import { Component } from "react";

import { TextField } from "../../components/TextField";
import i18n from "../../i18n";

const decorate = withStyles(theme => ({
  cardActions: {
    flexDirection: "row-reverse" as "row-reverse"
  },
  textField: {
    marginBottom: "2rem"
  }
}));

interface BaseCategoryFormProps {
  confirmButtonLabel: string;
  description: string;
  handleConfirm(data);
  name: string;
  title: string;
  errors: Array<{
    field: string;
    message: string;
  }>;
}

interface BaseCategoryFormState {
  formData: {
    description: string;
    name: string;
  };
}

export const BaseCategoryForm = decorate(
  class BaseCategoryForm extends Component<
    BaseCategoryFormProps & WithStyles<"cardActions" | "textField">,
    BaseCategoryFormState
  > {
    state = {
      formData: {
        description: this.props.description,
        name: this.props.name
      }
    };

    handleInputChange = event => {
      const { target } = event;
      this.setState(prevState => ({
        formData: { ...prevState.formData, [target.name]: target.value }
      }));
    };

    render() {
      const {
        classes,
        confirmButtonLabel,
        description,
        handleConfirm,
        name,
        title,
        errors
      } = this.props;
      const errorList = errors.reduce((acc, curr) => {
        acc[curr.field] = curr.message;
        return acc;
      }, {});
      console.log(errorList);
      return (
        <Card>
          <CardContent>
            <Typography variant="display1">{title}</Typography>
          </CardContent>
          <CardContent>
            <TextField
              autoFocus
              name="name"
              label={i18n.t("Name", { context: "category" })}
              defaultValue={name}
              className={classes.textField}
              onChange={this.handleInputChange}
              error={Boolean(errorList["name"])}
              helperText={errorList["name"]}
            />
            <TextField
              name="description"
              label={i18n.t("Description (optional)")}
              defaultValue={description}
              multiline
              onChange={this.handleInputChange}
              error={Boolean(errorList["description"])}
              helperText={errorList["description"]}
            />
          </CardContent>
          <CardContent>
            <CardActions className={classes.cardActions}>
              <Button
                variant="raised"
                color="primary"
                onClick={() => handleConfirm(this.state.formData)}
              >
                {confirmButtonLabel}
              </Button>
              <Button color="primary" onClick={() => window.history.back()}>
                {i18n.t("Cancel", { context: "button" })}
              </Button>
            </CardActions>
          </CardContent>
        </Card>
      );
    }
  }
);
