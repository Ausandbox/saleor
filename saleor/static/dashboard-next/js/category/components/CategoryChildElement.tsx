import * as React from "react";
import ButtonBase from "material-ui/ButtonBase";
import Card, { CardHeader } from "material-ui/Card";
import Paper from "material-ui/Paper";
import { withStyles } from "material-ui/styles";
import Typography from "material-ui/Typography";
import { Link } from "react-router-dom";
import Folder from "material-ui-icons/Folder";

import Skeleton from "../../components/Skeleton";

interface CategoryChildElementProps {
  label: string;
  url: string;
  loading?: boolean;
}

const decorate = withStyles(theme => ({
  button: {
    backgroundColor: theme.palette.background.paper,
    boxShadow: theme.shadows[2],
    color: theme.palette.getContrastText(theme.palette.background.paper),
    justifyContent: "flex-start" as "flex-start",
    transition: theme.transitions.create(["background-color", "box-shadow"], {
      duration: theme.transitions.duration.short
    }),
    width: "100%",
    "&$keyboardFocused": {
      boxShadow: theme.shadows[6]
    },
    "&:active": {
      boxShadow: theme.shadows[8]
    },
    "&$disabled": {
      boxShadow: theme.shadows[0],
      backgroundColor: theme.palette.action.disabledBackground
    },
    "&:hover": {
      backgroundColor: theme.palette.grey.A100,
      // Reset on mouse devices
      "@media (hover: none)": {
        backgroundColor: theme.palette.grey[300]
      },
      "&$disabled": {
        backgroundColor: theme.palette.action.disabledBackground
      }
    }
  }
}));

export const CategoryChildElement = decorate<CategoryChildElementProps>(
  ({ classes, label, url, loading }) => (
    <ButtonBase
      focusRipple
      className={classes.button}
      component={props => <Link to={url} {...props} />}
    >
      <CardHeader
        avatar={<Folder />}
        title={loading ? <Skeleton style={{ width: "80%" }} /> : label}
      />
    </ButtonBase>
  )
);

export default CategoryChildElement;
