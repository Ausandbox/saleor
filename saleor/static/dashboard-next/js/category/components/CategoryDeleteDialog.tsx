import Button from "material-ui/Button";
import Dialog, {
  DialogActions,
  DialogContent,
  DialogTitle
} from "material-ui/Dialog";
import Typography from "material-ui/Typography";
import { withStyles } from "material-ui/styles";
import * as React from "react";

import i18n from "../../i18n";

interface CategoryDeleteDialogProps {
  onClose?();
  onConfirm?();
  opened?: boolean;
}

export const CategoryDeleteDialog: React.StatelessComponent<
  CategoryDeleteDialogProps
> = props => {
  const { children, opened, onConfirm, onClose, ...dialogProps } = props;
  return (
    <Dialog open={opened} {...dialogProps}>
      <DialogTitle>
        {i18n.t("Delete category", { context: "title" })}
      </DialogTitle>
      <DialogContent>{children}</DialogContent>
      <DialogActions>
        <Button color="primary" onClick={onClose}>
          {i18n.t("Cancel", { context: "button" })}
        </Button>
        <Button color="primary" variant="raised" onClick={onConfirm}>
          {i18n.t("Delete category", { context: "button" })}
        </Button>
      </DialogActions>
    </Dialog>
  );
};
