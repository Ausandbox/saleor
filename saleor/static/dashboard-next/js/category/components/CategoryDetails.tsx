import Button from "material-ui/Button";
import Card, { CardActions, CardContent } from "material-ui/Card";
import Typography from "material-ui/Typography";
import * as React from "react";
import { Link } from "react-router-dom";

import { Skeleton } from "../../components/Skeleton";
import i18n from "../../i18n";

interface CategoryDetailsProps {
  description: string;
  editButtonLink: string;
  handleRemoveButtonClick?();
  loading: boolean;
  title: string;
}

export const CategoryDetails: React.StatelessComponent<
  CategoryDetailsProps
> = ({
  description,
  editButtonLink,
  handleRemoveButtonClick,
  loading,
  title
}) => (
  <Card>
    <CardContent>
      <Typography variant="headline" component="h2">
        {loading ? <Skeleton style={{ width: "10em" }} /> : title}
      </Typography>
      <Typography component="p">
        {loading
          ? [
              <Skeleton key="skel-1" style={{ width: "80%" }} />,
              <Skeleton key="skel-2" style={{ width: "75%" }} />,
              <Skeleton key="skel-3" style={{ width: "60%" }} />
            ]
          : description}
      </Typography>
    </CardContent>
    <CardActions>
      <Button
        color="primary"
        component={props => <Link to={editButtonLink} {...props} />}
      >
        {i18n.t("Edit", { context: "button" })}
      </Button>
      <Button color="primary" onClick={handleRemoveButtonClick}>
        {i18n.t("Delete", { context: "button" })}
      </Button>
    </CardActions>
  </Card>
);
