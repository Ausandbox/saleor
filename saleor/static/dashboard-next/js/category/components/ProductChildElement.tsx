import * as React from "react";
import ButtonBase from "material-ui/ButtonBase";
import Grid from "material-ui/Grid";
import Card, { CardContent, CardMedia } from "material-ui/Card";
import Typography from "material-ui/Typography";
import { Link } from "react-router-dom";
import { withStyles } from "material-ui/styles";
import { grey } from "material-ui/colors";
import Skeleton from "../../components/Skeleton";

interface CategoryChildElementProps {
  label: string;
  url: string;
  price: string;
  thumbnail: string;
  loading?: boolean;
}

const decorate = withStyles(theme => ({
  button: {
    backgroundColor: theme.palette.background.paper,
    boxShadow: theme.shadows[2],
    color: theme.palette.getContrastText(theme.palette.background.paper),
    display: "block",
    transition: theme.transitions.create(["background-color", "box-shadow"], {
      duration: theme.transitions.duration.short
    }),
    width: "100%",
    "&$keyboardFocused": {
      boxShadow: theme.shadows[6]
    },
    "&:active": {
      boxShadow: theme.shadows[8]
    },
    "&$disabled": {
      boxShadow: theme.shadows[0],
      backgroundColor: theme.palette.action.disabledBackground
    },
    "&:hover": {
      backgroundColor: theme.palette.grey.A100,
      // Reset on mouse devices
      "@media (hover: none)": {
        backgroundColor: theme.palette.grey[300]
      },
      "&$disabled": {
        backgroundColor: theme.palette.action.disabledBackground
      }
    }
  },
  media: {
    backgroundSize: "contain",
    height: 200
  },
  content: {
    height: "5rem"
  },
  price: {
    color: grey[500]
  }
}));

export const ProductChildElement = decorate<CategoryChildElementProps>(
  ({ label, price, url, thumbnail, classes, loading }) => (
    <ButtonBase
      focusRipple
      className={classes.button}
      component={props => <Link to={url} {...props} />}
    >
      <CardMedia image={thumbnail} className={classes.media} />
      <CardContent className={classes.content}>
        {loading ? (
          <Typography variant="button">
            <Skeleton style={{ width: "80%" }} />
          </Typography>
        ) : (
          <>
            <Typography variant="button">{label}</Typography>
            <Typography className={classes.price}>{price}</Typography>
          </>
        )}
      </CardContent>
    </ButtonBase>
  )
);

export default ProductChildElement;
