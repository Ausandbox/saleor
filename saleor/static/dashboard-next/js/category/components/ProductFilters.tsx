import * as React from "react";
import TextField from "material-ui/TextField";

import FilterCard, { FilterCardProps } from "../../components/cards/FilterCard";

export const ProductFilters: React.StatelessComponent<FilterCardProps> = ({
  handleClear,
  handleSubmit
}) => (
  <FilterCard handleClear={handleClear} handleSubmit={handleSubmit}>
    <TextField fullWidth name="name" placeholder="Name" />
  </FilterCard>
);

export default ProductFilters;
