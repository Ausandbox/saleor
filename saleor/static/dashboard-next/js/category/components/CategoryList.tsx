import * as React from "react";
import Typography from "material-ui/Typography";
import Grid from "material-ui/Grid";
import Button from "material-ui/Button";
import { Link } from "react-router-dom";

import CategoryChildElement from "./CategoryChildElement";
import { categoryAddUrl, categoryShowUrl } from "../";
import { CategoryPropertiesQuery } from "../gql-types";
import i18n from "../../i18n";

interface CategoryListProps {
  categories?: CategoryPropertiesQuery["category"]["children"]["edges"];
}
export const CategoryList: React.StatelessComponent<CategoryListProps> = ({
  categories
}) => (
  <Grid container>
    {categories === undefined ? (
      <Grid item xs={12} sm={6} md={4} lg={3} xl={2}>
        <CategoryChildElement loading={true} label="" url="" />
      </Grid>
    ) : categories.length > 0 ? (
      categories.map(edge => (
        <Grid item xs={12} sm={6} md={4} lg={3} xl={2}>
          <CategoryChildElement
            url={categoryShowUrl(edge.node.id)}
            label={edge.node.name}
            key={edge.node.id}
          />
        </Grid>
      ))
    ) : (
      <Grid item xs={12}>
        <Typography variant="body2">{i18n.t("No categories found")}</Typography>
      </Grid>
    )}
  </Grid>
);

export default CategoryList;
