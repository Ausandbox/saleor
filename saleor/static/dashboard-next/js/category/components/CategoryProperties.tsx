import { DialogContentText } from "material-ui/Dialog";
import * as React from "react";
import { Component } from "react";
import { Query } from "react-apollo";
import { Redirect } from "react-router-dom";

import { CategoryDeleteDialog } from "./CategoryDeleteDialog";
import { CategoryDetails } from "./CategoryDetails";
import {
  TypedCategoryDeleteMutation,
  categoryDeleteMutation
} from "../mutations";
import { pgettext, interpolate, ngettext } from "../../i18n";
import { categoryEditUrl, categoryShowUrl } from "../index";
import { CategoryPropertiesQuery } from "../gql-types";

interface CategoryPropertiesProps {
  loading: boolean;
  category?: CategoryPropertiesQuery["category"];
}

interface CategoryPropertiesState {
  opened: boolean;
}

export class CategoryProperties extends Component<
  CategoryPropertiesProps,
  CategoryPropertiesState
> {
  state = { opened: false };

  handleRemoveButtonClick = () => {
    this.setState(prevState => ({ opened: !prevState.opened }));
  };

  render() {
    const { category, loading } = this.props;
    const titleFmt = pgettext(
      "Remove category modal title",
      "Remove category %s"
    );
    const contentFmt = pgettext(
      "Remove category modal title",
      "Are you sure you want to remove category <strong>%s</strong>?"
    );
    return (
      <TypedCategoryDeleteMutation
        mutation={categoryDeleteMutation}
        variables={{ id: loading ? "" : category.id }}
      >
        {(deleteCategory, deleteResult) => {
          if (deleteResult && !deleteResult.loading) {
            this.handleRemoveButtonClick();
            return (
              <Redirect
                to={categoryShowUrl(
                  category.parent ? category.parent.id : null
                )}
                push={false}
              />
            );
          }

          return (
            <>
              <CategoryDetails
                description={loading ? "" : category.description}
                editButtonLink={loading ? "" : categoryEditUrl(category.id)}
                handleRemoveButtonClick={this.handleRemoveButtonClick}
                loading={loading}
                title={loading ? "" : category.name}
              />
              {!loading ? (
                <CategoryDeleteDialog
                  onClose={this.handleRemoveButtonClick}
                  onConfirm={() => deleteCategory()}
                  opened={this.state.opened}
                  title={interpolate(titleFmt, [category.name])}
                >
                  <DialogContentText
                    dangerouslySetInnerHTML={{
                      __html: interpolate(contentFmt, [category.name])
                    }}
                  />
                  {category.products &&
                    category.products.totalCount > 0 && (
                      <DialogContentText>
                        {interpolate(
                          ngettext(
                            "There is one product in this category that will be removed.",
                            "There are %s products in this category that will be removed.",
                            category.products.totalCount
                          ),
                          [category.products.totalCount]
                        )}
                      </DialogContentText>
                    )}
                </CategoryDeleteDialog>
              ) : null}
            </>
          );
        }}
      </TypedCategoryDeleteMutation>
    );
  }
}

export default CategoryProperties;
