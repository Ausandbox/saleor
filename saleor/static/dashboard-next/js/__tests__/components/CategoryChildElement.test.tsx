import * as React from "react";
import { MemoryRouter } from "react-router-dom";
import * as renderer from "react-test-renderer";

import { CategoryChildElement } from "../../category/components/CategoryChildElement";
import categoryListFixture from "./fixtures/categoryList";

const categoryFixture = categoryListFixture[0];

describe("<ProductChildElement />", () => {
  it("renders while data is loading", () => {
    const component = renderer.create(
      <MemoryRouter>
        <CategoryChildElement label="" url="" loading={true} />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
  it("renders when data is loaded", () => {
    const component = renderer.create(
      <MemoryRouter>
        <CategoryChildElement
          label={categoryFixture.node.name}
          url={`/categories/${categoryFixture.node.id}/`}
          loading={false}
        />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
