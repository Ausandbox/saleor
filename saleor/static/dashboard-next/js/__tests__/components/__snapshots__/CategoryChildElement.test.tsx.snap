// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductChildElement /> renders when data is loaded 1`] = `
<a
  className="MuiButtonBase-root-2 Component-button-1"
  href="/categories/Q2F0ZWdvcnk6MQ==/"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="MuiCardHeader-root-5"
  >
    <div
      className="MuiCardHeader-avatar-6"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-11"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
        />
      </svg>
    </div>
    <div
      className="MuiCardHeader-content-8"
    >
      <span
        className="MuiTypography-root-17 MuiTypography-body2-25 MuiCardHeader-title-9"
      >
        Apparel
      </span>
    </div>
  </div>
  <span
    className="MuiTouchRipple-root-42"
  />
</a>
`;

exports[`<ProductChildElement /> renders while data is loading 1`] = `
<a
  className="MuiButtonBase-root-2 Component-button-1"
  href="/"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="MuiCardHeader-root-5"
  >
    <div
      className="MuiCardHeader-avatar-6"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-11"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
        />
      </svg>
    </div>
    <div
      className="MuiCardHeader-content-8"
    >
      <span
        className="MuiTypography-root-17 MuiTypography-body2-25 MuiCardHeader-title-9"
      >
        <span
          className="Component-skeleton-41"
          style={
            Object {
              "width": "80%",
            }
          }
        >
          â€Œ
        </span>
      </span>
    </div>
  </div>
  <span
    className="MuiTouchRipple-root-42"
  />
</a>
`;
